### DISCUSSION TITLE
mkdocs macro plugin and creation of data classes from dictionaries?

### DISCUSSION THREAD

**Discussion by sslHello** [fralau/mkdocs-macros-plugin#101] [Q&A]
Hi,
I am writing a macro that uses tree dataclasses created from a dictionary (originally it comes from a YAML file).
Although a small test works OK in a 'normal' Python script:
[dict_2_dataclass.py](https://github.com/sslHello/mkdocs_dict_2_dataclass/blob/main/dict_2_dataclass.py)

... it does not work if the same code is called by the mkdocs_macros_plugin:
[dict_2_dataclass_mkdocs.py](https://github.com/sslHello/mkdocs_dict_2_dataclass/blob/main/dict_2_dataclass_mkdocs.py)

```
  File "/home/user/Top10/2021/dict_2_dataclass_mkdocs.py", line 25, in define_env
    result = from_dict(Y, d)
  File "/usr/local/lib/python3.9/dist-packages/dacite/core.py", line 46, in from_dict
    data_class_hints = get_type_hints(data_class, globalns=config.forward_references)
  File "/usr/lib/python3.9/typing.py", line 1408, in get_type_hints
    base_globals = sys.modules[base.__module__].__dict__
KeyError: 'dict_2_dataclass_mkdocs'
``` 
see [complete traceback](https://github.com/sslHello/mkdocs_dict_2_dataclass/blob/main/mkdocs_dict_2_dataclass_err.txt)

I've searched for any hint that can explain and solve this behaviour.  I haven't found anything yet ...
If you have any hint or idea for a solution I would greatly appreciate it.

**Comment by fralau** [comment]
🤔 

Could you try to break down your line 25 to smaller pieces,  to find exactly which step is failing?

**Reply by fralau** [reply]
@sslHello How are things going, with this question?
